lib/fastifySession.js

```diff
@@ -53,21 +57,24 @@ function session (fastify, options, next) {
         }
         return
       }
+
       if (!session) {
         request.session = new Session(request, idGenerator, cookieOpts, cookieSigner)
         done()
         return
       }
-      if (session.cookie?.expires && session.cookie.expires <= Date.now()) {
-        const restoredSession = new Session(
-          request,
-          idGenerator,
-          cookieOpts,
-          cookieSigner,
-          session,
-          decryptedSessionId
-        )
 
+      request.session = new Session(
+        request,
+        idGenerator,
+        cookieOpts,
+        cookieSigner,
+        session,
+        decryptedSessionId
+      )
```

Comment:
We might as well push this up.

It's pretty helpful because it converts `cookie.expires` into a date for us, so we can use `getTime()` for comparison against `Date.now()`.

Reply:
I think the issue is, that even the session maybe expired we have now a session assigned to the request?!

Reply:
`destroy()` already replaces the session in the request:
https://github.com/fastify/session/blob/master/lib/session.js#L87-L90

I could avoid setting request.session later though, if it makes you feel more comfortable. But it shouldn't make a difference, I think

Reply:
I guess this could be a problem if `destroy()` runs into an error.

I pushed a commit that assigns the session later. Thanks!